test/types/docArray.test.ts

```diff
@@ -24,3 +24,47 @@ async function gh10293() {
     return test.arrayOfArray; // <-- error here if the issue persisted
   };
 }
+
+function gh13087() {
+  interface Book {
+    author: {
+      name: string;
+    };
+  }
+
+  expectError(new Types.DocumentArray<Book>([1, 2, 3]));
+
+  const locationSchema = new Schema(
+    {
+      type: {
+        required: true,
+        type: String,
+        enum: ['Point']
+      },
+      coordinates: {
+        required: true,
+        type: [Number]
+      }
+    },
+    { _id: false }
+  );
+
+  const pointSchema = new Schema({
+    name: { required: true, type: String },
+    location: { required: true, type: String }
+  });
+
+  const routeSchema = new Schema({
+    points: { type: [pointSchema] }
+  });
+
+  type Route = InferSchemaType<typeof routeSchema>;
+
+  expectError(function getTestRouteData(): Route {
+    return {
+      points: new Types.DocumentArray([
+        { name: 'Test' } // "location" is missing
```

Comment:
The major problem with this approach is that you'll still get an error even if `location` has a default.

Even if we work around the defaults issue, still doesn't handle the case of custom getters, or properties that are set in middleware (e.g. timestamps).